### DISCUSSION TITLE
Is there a way to change default logger?

### DISCUSSION THREAD

**Discussion by flavacaster** [symfony/symfony#45065] [Q&A] [answered]
I'm using Symfony v6.0.2

I just want to know why this happens, don't want to use monolog just yet

I've tried this in my services.php:

`$services->set(\Psr\Log\LoggerInterface::class)
   ->class(\Symfony\Component\ErrorHandler\BufferingLogger::class);`
   
(Configuration like this worked when I specified implementation of HttpClientInterface)   
   
<img width="1331" alt="image" src="https://user-images.githubusercontent.com/54861683/150035104-aaf67795-ab6f-4f5f-8145-092a9f47c489.png">

But when I request  LoggerInterface in the constructor of TestCommand, I still get an instance of \Symfony\Component\HttpKernel\Log\Logger, and not \Symfony\Component\ErrorHandler\BufferingLogger

<img width="1488" alt="image" src="https://user-images.githubusercontent.com/54861683/150035307-33a638e5-cc5f-410a-9186-b5ca3293cb40.png">

I'd like to know if it's event possible to register some other implementation of LoggerInterface rather than the default

**Comment by nusje2000** [comment, marked as answer]
The LoggerInterface is registered as an alias for the "logger" service. I think by registering this service using the "logger" service ID you should be able to use the other implementation:

```php
$services->set('logger') ->class(\Symfony\Component\ErrorHandler\BufferingLogger::class);
```

**Reply by ghost** [reply]
Your solution is correct but let's explain how it works.
When you typehint an argument with LoggerInterface, a DI container must give you a service with id `LoggerInterface`. The OP correctly creates the service with id `LoggerInterface`. However problem exists later in container compilation process. The LoggerPass sets an alias `LoggerInterface` for a service `logger`. When creating an alias a definition with the same id is removed if exists. This way a definition created by OP gets removed.

**Reply by stof** [reply]
The convention in Symfony since the 2.0.0 release is that the default logger is configured as the `logger` service.
When adding autowiring in Symfony 3.3, we added an autowiring alias for the LoggerInterface, so that autowiring the LoggerInterface injects that `logger` service.

so if you want to configure the default logger, you should configure that `logger` service (which will then also be taken into account by all core services that don't rely on autowiring and inject the logger using the `logger` id directly).
